Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Fixes/ItemFlagsInSwitchFixes.cs

```diff
@@ -0,0 +1,129 @@
+    {
+        public const string FixKeyConvertItemFlagsSwitch = "ConvertItemFlagsSwitch";
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            Constants.DiagnosticPrefix + "003"
+        );
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken)
+                .ConfigureAwait(false);
+            if (root == null)
+            {
+                return;
+            }
+
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan span = diagnostic.Location.SourceSpan;
+            CaseSwitchLabelSyntax? caseLabel = root.FindToken(span.Start).Parent?
+                .FirstAncestorOrSelf<CaseSwitchLabelSyntax>();
+            if (caseLabel == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: "Convert case to use HasFlag",
+                    createChangedDocument: c => ConvertCaseToHasFlag(
+                        document: context.Document,
+                        caseLabel: caseLabel,
+                        cancellationToken: c
+                    ),
+                    equivalenceKey: FixKeyConvertItemFlagsSwitch
+                ),
+                diagnostic
+            );
+        }
+
+        private async Task<Document> ConvertCaseToHasFlag(
+            Document document,
+            CaseSwitchLabelSyntax caseLabel,
+            CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+            SwitchSectionSyntax switchSection = (SwitchSectionSyntax)caseLabel.Parent!;
+            SwitchStatementSyntax switchStatement = (SwitchStatementSyntax)switchSection.Parent!;
+            ExpressionSyntax switchExpression = switchStatement.Expression;
+            
+            string variableName = GetPatternMatchingVariableName(editor, switchStatement);
+
+            // Generate the pattern matching case label
+            CasePatternSwitchLabelSyntax newCaseLabel = GeneratePatternMatchingCaseLabel(caseLabel, switchExpression, variableName);
+
+            // Replace the old case label with the new case label
+            editor.ReplaceNode(caseLabel, newCaseLabel);
+
+            Document newDoc = editor.GetChangedDocument();
+            return newDoc;
+        }
+
+        private static string GetPatternMatchingVariableName(DocumentEditor editor, SwitchStatementSyntax switchStatement)
+        {
+            var analysis = editor.SemanticModel.AnalyzeDataFlow(switchStatement);
+
+            if (analysis is null)
+            {
+                return "f";
+            }
+            
+            var hasCollision = analysis.WrittenInside.Any(x => x.Name.StartsWith("f"));
```

Comment:
In keeping with the above comment, I think you may be looking for CapturedInside, CapturedOutside, or DataFlowsIn (to see what all variables are available to the scope). I think we also need a test which has an unrelated `f` defined outside the switch to make sure we avoid that scoping collision

Reply:
`WrittenInside` was one of the only ones actually showing the `f` variable in it. I hadn't considered an unrelated `f`, though, which will likely change things.